From 1bd0ecee0cf4f32383c412f12c6fcbe834ac0f13 Mon Sep 17 00:00:00 2001
From: Narendhar Reddy <quic_nare@quicinc.com>
Date: Wed, 10 Jan 2024 16:36:31 +0530
Subject: [PATCH] wifi: mac80211: add WQ_MEM_RECLAIM flag to wiphy workqueue.

When a workqueue with WQ_MEM_RECLAIM tries to flush the workqueue
without the flag WQ_MEM_RECLAIM it throws following warning.

Add the WQ_MEM_RECLAIM flag to avoid this.

workqueue: WQ_MEM_RECLAIM cfg80211:0xffffffc00943c248 is flushing !WQ_MEM_RECLAIM 
phy3:ath12k_wmi_peer_chan_width_switch_work [ath12k]
WARNING: CPU: 3 PID: 32563 at kernel/workqueue.c:2623 check_flush_dependency+0xec/0x100

Signed-off-by: Narendhar Reddy <quic_nare@quicinc.com>
---
 net/mac80211/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/main.c b/net/mac80211/main.c
index 31e9254..78fc1ce 100644
--- a/net/mac80211/main.c
+++ b/net/mac80211/main.c
@@ -1416,7 +1416,7 @@ int ieee80211_register_hw(struct ieee80211_hw *hw)
 		hw->queues = IEEE80211_MAX_QUEUES;
 
 	local->workqueue =
-		alloc_ordered_workqueue("%s", 0, wiphy_name(local->hw.wiphy));
+		alloc_ordered_workqueue("%s", WQ_MEM_RECLAIM, wiphy_name(local->hw.wiphy));
 	if (!local->workqueue) {
 		result = -ENOMEM;
 		goto fail_workqueue;
-- 
2.34.1

